controllers/articles-controllers.js

```diff
@@ -0,0 +1,13 @@
+const {readArticleById} = require("../models/articles-models")
+
+exports.getArticleById = (req, res, next) => {
+    const {article_id} = req.params 
+
+    readArticleById(article_id).then((article) => {
+        if (article) res.status(200).send({article})
+        else next({status: 404, msg: "Article not found"})
+    })
```

Comment:
I would deal with this in your model, check the rows in the .then block in the model and use promise reject. Then you already have a catch block set up here to catch the error and send it to next

Controllers should mostly just focus on dealing with requests, responses and catching errors.

It does work though!